src/test/enginesynctest.cpp

```diff
@@ -1051,61 +1054,69 @@ TEST_F(EngineSyncTest, MomentarySyncDependsOnPlayingStates) {
 }
 
 TEST_F(EngineSyncTest, EjectTrackSyncRemains) {
-    auto pButtonMasterSyncInternal = std::make_unique<ControlProxy>(m_sInternalClockGroup, "sync_master");
     auto pButtonSyncEnabled1 = std::make_unique<ControlProxy>(m_sGroup1, "sync_enabled");
     auto pButtonSyncEnabled2 = std::make_unique<ControlProxy>(m_sGroup2, "sync_enabled");
     auto pButtonEject1 = std::make_unique<ControlProxy>(m_sGroup1, "eject");
 
-    pButtonMasterSyncInternal->set(1.0);
-
-    auto pFileBpm1 = std::make_unique<ControlProxy>(m_sGroup1, "file_bpm");
-    pFileBpm1->set(120.0);
+    BeatsPointer pBeats1 = BeatFactory::makeBeatGrid(*m_pTrack1, 120, 0.0);
+    m_pTrack1->setBeats(pBeats1);
     pButtonSyncEnabled1->set(1.0);
-
     ProcessBuffer();
+
+    // m_sGroup1 takes over
+    assertIsFollower(m_sInternalClockGroup);
+    assertIsSoftMaster(m_sGroup1);
+    assertSyncOff(m_sGroup2);
+
     pButtonEject1->set(1.0);
     // When an eject happens, the bpm gets set to zero.
-    pFileBpm1->set(0.0);
     ProcessBuffer();
 
-    // Group1 is still master because there are no other sync decks.
-    assertIsSoftMaster(m_sGroup1);
+    EXPECT_FLOAT_EQ(0.0,
+            ControlObject::getControl(ConfigKey(m_sGroup1, "bpm"))->get());
+
+    // No valid tempo available all are waiting as follower
+    assertIsFollower(m_sInternalClockGroup);
+    assertIsFollower(m_sGroup1);
+    assertSyncOff(m_sGroup2);
 
-    // Try again with multiple sync decks
-    pFileBpm1->set(128.0);
-    auto pFileBpm2 = std::make_unique<ControlProxy>(m_sGroup2, "file_bpm");
-    pFileBpm2->set(135.0);
+    m_pMixerDeck1->loadFakeTrack(false, 128.0);
+    EXPECT_FLOAT_EQ(128.0,
+            ControlObject::getControl(ConfigKey(m_sGroup1, "bpm"))->get());
+    BeatsPointer pBeats2 = BeatFactory::makeBeatGrid(*m_pTrack2, 135, 0.0);
+    m_pTrack2->setBeats(pBeats2);
     pButtonSyncEnabled2->set(1.0);
     ProcessBuffer();
 
-    assertIsSoftMaster(m_sInternalClockGroup);
+    // assertIsSoftMaster(m_sInternalClockGroup);
```

Comment:
remove?  what is the internal clock here?

Reply:
Soft master. This was only a debug leftover. I will uncomment it.